lib/msal-core/src/XHRClient.ts

```diff
@@ -31,7 +32,10 @@ export class XhrClient {
                     reject(this.handleError(xhr.responseText));
                 }
 
-                resolve(jsonResponse);
+                resolve({
+                    client: xhr,
```

Comment:
We had talked (in the context of the throttling work) that this would return a few known properties, such as headers, status, and body. I would prefer that over returning the entire `xhr` object.

Reply:
And you shuold add a new type to represent this object, instead of `any`.

Reply:
Added  XhrResponse type with responseBody, and statusCode can add other valuable attributes as needed